libpoolprotocols/stratum/EthStratumClient.cpp

```diff
@@ -1257,6 +1463,53 @@ void EthStratumClient::processResponse(Json::Value& responseObject)
                     processExtranonce(enonce);
             }
         }
+        else if (_method == "mining.set" && m_conn->StratumMode() == ETHEREUMSTRATUM2)
+        {
+            /*
+            {
+              "method": "mining.set",
+              "params": {
+                  "epoch" : "dc",
+                  "target" : "0112e0be826d694b2e62d01511f12a6061fbaec8bc02357593e70e52ba",
+                  "algo" : "ethash",
+                  "extranonce" : "af4c"
+              }
+            }
+            */
+            if (!responseObject.isMember("params") || !responseObject["params"].isObject() ||
+                responseObject["params"].empty())
+            {
+                cwarn << "Got invalid mining.set message. Discarding ...";
+                return;
+            }
+            m_session->firstMiningSet = true;
+            jPrm = responseObject["params"];
+            string timeout = jPrm.get("timeout", "").asString();
+            string epoch = jPrm.get("epoch", "").asString();
+            string target = jPrm.get("target", "").asString();
+
+            if (!timeout.empty())
+                m_session->timeout = stoi(timeout, nullptr, 16);
+
+            if (!epoch.empty())
+                m_session->epoch = stoul(epoch, nullptr, 16);
+
+            if (!target.empty())
+            {
+                target = "0x" + dev::padLeft(target, 64, '0');
+                m_session->nextWorkBoundary = h256(target);
+            }
+
+            m_session->algo = jPrm.get("algo", "ethash").asString();
```

Comment:
Is the `.algo` ever checked?
I also would propose to use `ethash/1.0.0` instead of just `ethash`, because we might need the version for `progpow`.

Reply:
Not yet ... nevertheless algo is a member of the mining.set method.
I will later implement a parser for versioning.
This is preparatory work.